File: previous year/1styear/python/theory/sonal/quick_sort.py
def partition(b, h, k):
	i = h 
	j = k
	pivot = b[h]

	while i < j:
		if pivot > b[i+1]:
			# print(i)
			b[i + 1], b[i] = b[i], b[i + 1]
			i += 1

		else:
			# print(i,j)
			b[i+1], b[j] = b[j], b[i+1]
			j -= 1
	return i

def quick_sort(b, h, k):
	
	if k - h < 1:
		return 

	j = partition(b, h, k)
	print(b,j)

	quick_sort(b, h, j - 1)
	quick_sort(b, j + 1, k)


lst = [4,7,2,1,5,3]
quick_sort(lst, 0, len(lst)-1)
print(lst)

class A:
	def __init__(self):
		print('dgrgrsrth')
class B(A):
	pass
# class C(B):
	# pass
c = B()

File: previous year/2ndYear/sem4/ajava/assignment_package/Sorting/Quick_sort.java
package Sorting;

public class Quick_sort {

    int partition(int arr[], int low, int high){
        // First element as pivot
        int pivot = arr[low];
        int st = low;
        int end
        = high;
        int k = high;
        for (int i = high; i > low; i--) {
        if (arr[i] > pivot)
        swap(arr, i, k--);
        }
        swap(arr, low, k);
        
        return k;
    
    }
    
    public static void swap(int[] arr, int i, int j){
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }
    
    public void quickSort(int arr[], int low, int high){
        if (low < high) {
        int idx = partition(arr, low, high);
        quickSort(arr, low, idx - 1);
        quickSort(arr, idx + 1, high);
        }
    }
    
    void printArray(int arr[], int size){
        int i;
        for (i = 0; i < size; i++)
        System.out.print(arr[i] + " ");
        System.out.println();
    }
    
}